app/models/collection_item.rb

```diff
@@ -0,0 +1,25 @@
+class CollectionItem < ApplicationRecord
+  belongs_to :book
+  belongs_to :user
+
+  validates :current_page, presence: true
+  validates :book_id, 
+            uniqueness: { scope: :user_id,
+                          message: "can only be added once in collection"
+                        } 
+
+  def book_data
```

Comment:
Can you elaborate why do you need a collection item between users and books?

Reply:
Thanks for this.

Ok The app is a BookStore CMS. We can think of the `Book` as the official supplier of a particular book and `CollectionItem` as the copy owned by a user. So the `admin` can buy more Books from the supplier and put in the app's library. Meanwhile, an ordinary user can search the library (Books already purchased from the supplier) and add a particular book in his/her collection (CollectionItem). 

The `Book` itself comes from a 3rd party API (which is called in the frontend - Googlebooks API).

I hope I was able to clarify it :)